services/rest_service.go

```diff
@@ -246,7 +253,9 @@ func unmarshalError(body []byte) (*models.DataEnvelope, error) {
 	var d models.DataEnvelope
 	err := json.Unmarshal(body, &d)
 	if err != nil {
-		return nil, err
+		// instead of bubbling the unmarshall error, let us return back the actual message so that it is not lost
+		d.Data.Message = fmt.Sprintf("unmarshall failed with error %s, this is the actual body: %s", err.Error(), string(body))
+		return &d, nil
```

Comment:
Where can we view d.Data.Message (i.e. where is it surfaced)?

Could another option also be returning the error which contains both the original error message as well as the `string(body)`?

Reply:
Ah, I had a closer look and we're just surfacing an error; so this is an error from an error. What you have makes sense then 👍🏼